lib/console_web/controllers/flows_controller.ex

```diff
@@ -33,11 +33,49 @@ defmodule ConsoleWeb.FlowsController do
               end
             end)
             |> Map.put(:organization_id, current_organization.id)
-            |> put_timestamps()
           end)
 
-        { count, _ } = Repo.insert_all(Flow, parsed_flows, on_conflict: :nothing)
-        if count == length(parsed_flows) do
+        flows_to_delete = existing_flows
+          |> Enum.filter(fn f ->
+            existing_flow = Map.take(f, [:device_id, :label_id, :function_id, :channel_id, :organization_id])
+
+            !Enum.any?(new_flows, fn new_flow ->
+              Map.equal?(new_flow, existing_flow)
+            end)
+          end)
+
+        flows_to_add = new_flows
+          |> Enum.filter(fn new_flow ->
+            parsed_existing_flows =
+              existing_flows
+              |> Enum.map(fn f -> Map.take(f, [:device_id, :label_id, :function_id, :channel_id, :organization_id]) end)
+
+            !Enum.any?(parsed_existing_flows, fn existing_flow ->
+              Map.equal?(new_flow, existing_flow)
+            end)
+          end)
+
+        {:ok, %{ flows_to_add: flows_to_add, flows_to_delete: flows_to_delete }}
+      end)
+      |> Ecto.Multi.run(:deleted_flows, fn _repo, %{ flows_changes: flows_changes } ->
+        ids_to_delete =
+          flows_changes.flows_to_delete |> Enum.map(fn f -> f.id end)
+
+        { count, nil } = from(f in Flow, where: f.id in ^ids_to_delete) |> Repo.delete_all()
+        {:ok, count}
+      end)
+      |> Ecto.Multi.run(:added_flows, fn _repo, %{ flows_changes: flows_changes } ->
+        flows_to_insert =
+          flows_changes.flows_to_add |> Enum.map(fn f ->
+            if f.device_id != nil, do: Devices.get_device!(current_organization, f.device_id)
```

Comment:
make sure that hackers dont connect resources from other orgs to the current org